[Enhancement] Reintroduce the `MapStore` interface and `SimpleMap` and make the `Badger` a submodules (#33)
pokt-network/smt: docs/smt.md

## Database

> By default this library provides a simple interface (`MapStore`) and a simple
> in-memory key-value database found in [`simplemap.go`](../kvstore/simplemap.go).
> However, any key-value store that implements this interface can be used as the
> node store to back the trie.

Reviewer: ```suggestion
By default, this library provides a simple interface (`MapStore`) and a simple
in-memory key-value database found in [`simplemap.go`](../kvstore/simplemap.go).
However, any key-value store implementing this interface can be used as the
node store to back the trie.
```

See: [`kvstore/simplemap.go`](../kvstore/simplemap.go) for the `MapStore`
interface and simple key-value map implementation.

### Database Submodules

> In addition to providing the `MapStore` and `SimpleMap` the `smt` library also
> provides wrappers around other key-value databases as submodules with more
> fully featured interfaces that can be used as outside of being a node store for
> the tries. These submodules can be found in the [`kvstore`](../kvstore/)
> directory.

Reviewer: ```suggestion
In addition to providing the `MapStore` and `SimpleMap` interfaces and
implementations, the `smt` library also provides wrappers around other
key-value databases as submodules with more fully-featured interfaces
that can be used outside of backing key-value engines for tries. These
submodules can be found in the [`kvstore`](../kvstore/) directory.
```

## Database

By default, this library provides a simple interface (`MapStore`) which can be
found in [`kvstore/interfaces.go`](../kvstore/interfaces.go) and submodule
implementations of said interface. These submodules allow for more extensible
key-value store implementations that give the user more control over their
database backing the underlying trie.

### Database Submodules

In addition to providing the `MapStore` interface and `simplemap`
implementation, the `smt` library also provides wrappers around other key-value
databases as submodules with more fully-featured interfaces that can be used
outside of backing key-value engines for tries. These submodules can be found in
the [`kvstore`](../kvstore/) directory.

#### SimpleMap

This library defines the `SimpleMap` interface which is implemented as an
extremely simple in-memory key-value store.

> Although it is is a submodule, it is ideal for simple, testing or non-production

Reviewer: ```suggestion
Although it is a submodule, it is ideal for simple, testing or non-production
```
